src/wix/WixToolset.Converters/WixConverter.cs

```diff
@@ -2854,39 +2862,168 @@ private static void RenameElementToStandardDirectory(XElement element)
+                                            bool protectQuotes = true,
+                                            bool reportWhitespace = false,
+                                            List<XNode> initialComments = null)
         {
             value = null;
-            comments = null;
-            var found = false;
-
+            char[] whitespaceChars = { ' ', '\t', '\r', '\n' };
             var nodes = element.Nodes().ToList();
-            comments = initialComments;
-            var nonCommentNodes = new List<XNode>();
+            comments = initialComments ?? new List<XNode>(); ;
+            char? currentQuote = null;
+            var inWhitespace = false;
+            var cDataFound = false;
+            var whitespaceFound = false;
+            var sb = new StringBuilder();
 
             foreach (var node in nodes)
             {
                 if (XmlNodeType.Comment == node.NodeType)
                 {
                     comments.Add(node);
                 }
-                else
+                else if (XmlNodeType.CDATA == node.NodeType || XmlNodeType.Text == node.NodeType)
                 {
-                    nonCommentNodes.Add(node);
+                    var isCData = XmlNodeType.CDATA == node.NodeType;
+
+                    if (isCData)
+                    {
+                        cDataFound = true;
+                    }
+
+                    var text = (node as XText)?.Value;
+
+                    if (protectQuotes)
+                    {
+                        var nodeSB = new StringBuilder();
+
+                        foreach (var c in text)
+                        {
+                            char? emit = c;
+
+                            // Replace contiguous whitespace with a single space.
+                            if (' ' == c || '\r' == c || '\n' == c || '\t' == c)
+                            {
+                                if (!inWhitespace)
+                                {
+                                    inWhitespace = true;
+                                    whitespaceFound = true;
+                                    emit = ' ';
+                                }
+                                else
+                                {
+                                    emit = null;
+                                }
+                            }
+                            else
+                            {
+                                inWhitespace = false;
+
+                                if ('\\' == c)
+                                {
+                                    // Escape backslash.
+                                    nodeSB.Append('\\');
+                                }
+
+                                else if (!isCData)
+                                {
+                                    // Escape nested quote.
+                                    if ('\'' == c || '"' == c)
```

Comment:
Single and double quote are not escaped using backslashes in XML. Is this getting used for something else? Since we're using an XAttribute to set the value, we don't have to escape anything. In fact, escaping would cause values to be double escaped.

Am I missing why these need to be backslash escaped?